integrations/spark-extensions-base/src/main/scala/org/apache/spark/sql/catalyst/parser/extensions/NessieSparkSqlExtensionsParser.scala

```diff
@@ -116,7 +116,17 @@ class NessieSparkSqlExtensionsParser(delegate: ParserInterface)
   }
 
   private def isNessieCommand(sqlText: String): Boolean = {
-    val normalized = sqlText.toLowerCase(Locale.ROOT).trim()
+    val normalized = sqlText
```

Comment:
It looks frustrating to have to use regex in order to "pre-parse" something that can be parsed with a proper Antlr grammar. Shouldn't we get rid of `isNessieCommand`?

Reply:
BTW I could find at least one statement that is correctly parsed by the grammar, but fails to be parsed by the regexes:

```
/* leading /* leading */ */ CREATE BRANCH IF NOT EXISTS %s_other IN nessie
```

Reply:
It's the same approach as in Iceberg